src/validation.h

```diff
@@ -798,6 +800,33 @@ class CChainState
 
     bool LoadBlockIndexDB() EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 
+    //! Indirection necessary to make lock annotations work with an optional mempool.
+    RecursiveMutex* MempoolMutex() const LOCK_RETURNED(m_mempool->cs)
+    {
+        return m_mempool ? &m_mempool->cs : nullptr;
+    }
+
+    /**
+     * Make mempool consistent after a reorg, by re-adding or recursively erasing
+     * disconnected block transactions from the mempool, and also removing any
+     * other transactions from the mempool that are no longer valid given the new
+     * tip/height.
+     *
+     * Note: we assume that disconnectpool only contains transactions that are NOT
+     * confirmed in the current chain nor already in the mempool (otherwise,
+     * in-mempool descendants of such transactions would be removed).
+     *
+     * Passing fAddToMempool=false will skip trying to add the transactions back,
+     * and instead just erase from the mempool as needed.
+     */
+    void TryUpdateMempoolForReorg(
```

Comment:
This new name seems worse to me than the original (`UpdateMempoolForReorg()`). The `Try` suggests that this operation may fail or that it's best-effort. In fact, if a mempool exists then it really will be updated to be internally consistent.

Reply:
If you felt this way it would have been helpful for you to chime in on @naumenkogs's original comment; doing commits back and forth on relatively cosmetic things like method names is not a good use of energy.

Reply:
Sorry - I didn't see the original comment. Ultimately it's up to you what you put in your PR. I don't think that renaming this function is necessary, and is actually worse for the reasons I gave here, but if other people prefer the new name, that's also fine!